From a81416a3b1b5b27c05f42f15a716c7623c405e34 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Date: Wed, 22 Apr 2020 22:28:58 +0200
Subject: [PATCH] cmake: fix check for memfd_create
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

memfd_create is disabled on uClibc:
https://cgit.uclibc-ng.org/cgi/cgit/uclibc-ng.git/commit/?id=0bbc6a4bf54197ec5dab874750d125cbd5b9e877

Instead of checking for the header file we need to check for
memfd_create(), fixes build error

xbmc/utils/UDMABufferObject.cpp:94:13: error:
 ‘memfd_create’ was not declared in this scope

for this kodi config:

-DCORE_PLATFORM_NAME=wayland
-DWAYLAND_RENDER_SYSTEM=gl

-- Core system type: linux
-- Platform: wayland
---
 cmake/scripts/linux/ArchSetup.cmake | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/cmake/scripts/linux/ArchSetup.cmake b/cmake/scripts/linux/ArchSetup.cmake
index acd62579817c..04fc6e3ffd1a 100644
--- a/cmake/scripts/linux/ArchSetup.cmake
+++ b/cmake/scripts/linux/ArchSetup.cmake
@@ -84,13 +84,6 @@ endif()
 include(LDGOLD)
 
 include(CheckIncludeFiles)
-check_include_files("linux/memfd.h" HAVE_LINUX_MEMFD)
-if(HAVE_LINUX_MEMFD)
-  list(APPEND ARCH_DEFINES "-DHAVE_LINUX_MEMFD=1")
-else()
-  message(STATUS "include/linux/memfd.h not found")
-endif()
-
 check_include_files("linux/udmabuf.h" HAVE_LINUX_UDMABUF)
 if(HAVE_LINUX_UDMABUF)
   list(APPEND ARCH_DEFINES "-DHAVE_LINUX_UDMABUF=1")
@@ -120,6 +113,15 @@ if(HAVE_MKOSTEMP)
   list(APPEND ARCH_DEFINES "-DHAVE_MKOSTEMP=1")
 endif()
 
+set(CMAKE_REQUIRED_DEFINITIONS "-D_GNU_SOURCE")
+check_symbol_exists("memfd_create" "sys/mman.h" HAVE_LINUX_MEMFD)
+set(CMAKE_REQUIRED_DEFINITIONS "")
+if(HAVE_LINUX_MEMFD)
+  list(APPEND ARCH_DEFINES "-DHAVE_LINUX_MEMFD=1")
+else()
+  message(STATUS "memfd_create() not found")
+endif()
+
 # Additional SYSTEM_DEFINES
 list(APPEND SYSTEM_DEFINES -DHAS_POSIX_NETWORK -DHAS_LINUX_NETWORK)
 
